src/lib/WebSocketClient.ts

```diff
@@ -0,0 +1,169 @@
+/* eslint-disable class-methods-use-this */
+import { v4 as genUuid } from 'uuid';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import type {
+  IMiddlewareWebSocketSubscriptionMessage,
+  ITopHeader,
+  ITransaction,
+  WebSocketChannelName,
+} from '../types';
+import {
+  WEB_SOCKET_CHANNELS,
+  WEB_SOCKET_SUBSCRIBE,
+  WEB_SOCKET_UN_SUBSCRIBE,
+  WEB_SOCKET_RECONNECT_TIMEOUT,
+  handleUnknownError,
+} from '../popup/utils';
+
+let wsClient: WebSocket;
+let isWsConnected: boolean;
+
+const subscribersQueue: IMiddlewareWebSocketSubscriptionMessage[] = [];
+
+const subscribers: Record<WebSocketChannelName, Record<string, (
+  payload: ITransaction | ITopHeader
+) => void>> = {
+  [WEB_SOCKET_CHANNELS.Transactions]: {},
+  [WEB_SOCKET_CHANNELS.MicroBlocks]: {},
+  [WEB_SOCKET_CHANNELS.KeyBlocks]: {},
+  [WEB_SOCKET_CHANNELS.Object]: {},
+};
+
```

Comment:
You choose to use a Singleton class so the states used within the class should be encapsulated inside of the class.